package.json

```diff
@@ -23,7 +23,22 @@
     },
     "devDependencies": {
         "dtslint": "latest",
+        "husky": ">=1",
+        "lint-staged": ">=8",
+        "prettier": "^1.17.1",
         "types-publisher": "github:Microsoft/types-publisher#production"
     },
-    "dependencies": {}
+    "dependencies": {},
+    "husky": {
+        "hooks": {
+            "pre-commit": "lint-staged",
+            "pre-push": "echo '\n[DefinitelyTyped] Running tests...\n' && yarn test"
```

Comment:
1. I don't think this is appropriate, for a couple of reasons:
  a. The tests are slow and the test infrastructure is hard to get installed correctly. I need to fix this, but it's in a bad state now.
  b. Some people are contributing minor changes that are low-priority to them. For them, it is fine to have a slow trip to CI to find out if they have errors. They'll never have more than 1 or 2 anyway.
1. Does this command work on Windows machines? I can't remember how cmd interprets quotes and `&&`. 
2. Can you use npm instead of yarn? I don't want to assume that yarn is installed.

I think we should re-evaluate this after
1. making the current infrastructure more stable.
2. creating a fast-run mode for tests to use here -- perhaps just linting each of the changed packages, ignoring dependents.

Reply:
Sounds good, for now I'll keep prettier in the precommit and remove prepush. We can return to this later.